app/models/king.rb

```diff
@@ -1,5 +1,48 @@
 class King < Piece
 
+  def valid_move?(new_x, new_y)
+    #checks if the move that the king is making is valid
+    if position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
+      return true
+    else
+      return false
+    end
+  end
+
+  def position_exist?(new_x, new_y)
+    #checks if position that king is moving to exists
+    if new_x < 0 || new_x > 7 || new_y < 0 || new_y > 7
+      return false
+    else
+      return true
+    end
```

Comment:
Would it be easier for the logic to flow like x cord is between 0 & 7  vs if it is not less than 0 or greater than 7?

``` ruby
def position_exist?(new_x, new_y)
  new_x >= 0 && new_x <= 7 && new_y >= 0 && new_y <= 7
end

Just thoughts after writing knight logic! :)